src/test/mds/TestMDSAuthCaps.cc

```diff
@@ -100,6 +100,32 @@ TEST(MDSAuthCaps, ParseGood) {
   }
 }
 
+TEST(MDSAuthCaps, ParseDumpReparseCaps) {
+  for (auto str : parse_good) {
+    MDSAuthCaps cap1;
+    ASSERT_TRUE(cap1.parse(str, &cout));
+
+    std::cout << "Testing by parsing caps, dumping to string, reparsing "
+		 "string and then redumping and checking strings from "
+		 "first and second dumps: '" << str << "'" << std::endl;
+    // Convert cap object to string, reparse and check if converting again
+    // gives same string as before.
+    MDSAuthCaps cap2;
+    std::ostringstream cap1_ostream;
+    cap1_ostream << cap1;
+    string cap1_str = cap1_ostream.str();
+    // Removing "MDSAuthCaps[" from cap1_str
+    cap1_str.replace(0, 12, "");
+    // Removing "]" from cap1_str
+    cap1_str.replace(cap1_str.length() - 1, 1, "");
+    ASSERT_TRUE(cap2.parse(cap1_str, &cout));
```

Comment:
Wouldn't it suffice to compare `cap1_str` with `str`?

Reply:
Yes, but the intention of this test is to ensure that the string obtained after parsing and reparsing is same. Using `str` directly will be against the idea of this testcase.

Reply:
if the strings match, and `str` parsing was already successful, then doesn't that imply that parsing it again would work.

Reply:
spoke to @rishabh-d-dave over a call - some redundancy can be removed.

Reply:
This doesn't work because if str is `allow rw fsname=a path="/foo"`, cap1_str will be `allow rw fsname=a path=/foo`. And therefore check if both are equal will always fail.

Reply:
Wouldn't excluding quotes from the path component suffice?

Reply:
Exclude from `str` or `cap_str`?

Reply:
In the example you have, from the original string.

Reply:
```
Testing good input: 'allow r ,allow * uid=1'
allow r ,allow * uid=1
allow r, allow * uid=1
/home/rishabh/repos/ceph/mdscaps/src/test/mds/TestMDSAuthCaps.cc:132: Failure
Value of: cap_str.compare(str) == 0
  Actual: false
Expected: true
[  FAILED  ] MDSAuthCaps.ParseDumpCompareCaps (0 ms)
```

Reply:
oh, the commas ;P